main/src/hadd.cxx

```diff
@@ -427,10 +427,10 @@ int main( int argc, char **argv )
          else
             newcomp = ROOT::RCompressionSetting::EDefaults::kUseCompiledDefault;
          delete firstInput;
-         fileMerger.SetMergeOptions(TString("first_source_compression"));
+         fileMerger.SetMergeOptions(TString("rnt:FirstSrcCompression"));
       } else {
          newcomp = ROOT::RCompressionSetting::EDefaults::kUseCompiledDefault;
-         fileMerger.SetMergeOptions(TString("default_compression"));
+         fileMerger.SetMergeOptions(TString("rnt:DefaultCompression"));
```

Comment:
That options is also not `RNTuple` specific.  The user requested the 'default' compression, this is true through-out.  The fact that everything else, 'just' setting the request compression to the global default means that until now pushing this information down was redundant but now it is not redundant anymore (no way to know if `newcomp` is set to the default by happenstance or intentionally).  The fact that only `RNTuple` will (for now?) use this information does not make it `RNTuple` specific.